packages/framework/aqueduct/src/container-runtime-factories/baseContainerRuntimeFactory.ts

```diff
@@ -97,18 +98,49 @@ export class BaseContainerRuntimeFactory
 		this.registry = new FluidDataStoreRegistry(this.registryEntries);
 	}
 
-	public async instantiateFirstTime(runtime: ContainerRuntime): Promise<void> {
+	/**
+	 * @deprecated - Use the overload taking in an IContainerRuntime
+	 */
+	// eslint-disable-next-line import/no-deprecated -- ContainerRuntime class to be moved to internal scope
+	public async instantiateFirstTime(runtime: ContainerRuntime): Promise<void>;
+	public async instantiateFirstTime(runtime: IContainerRuntime): Promise<void>;
```

Comment:
Can we document the new overload we consumers to use?

Reply:
(Same for the other replacement functions)

Reply:
Sure but first I have to figure out what they do! 😅